pieman.sh

```diff
@@ -183,48 +212,114 @@ run_scripts "bootstrap"
+    info "${IMAGE} of size ${image_size}M was successfully created"
+
+    info "creating loop device and scanning partition table"
+
+    scan_partition_table
+
+    info "${LOOP_DEV} is successfully created"
+
+    info "formatting partitions"
+
+    format_partitions vfat ext4
+
+    mount "${LOOP_DEV}p1" "${MOUNT_POINT}"
+
+    rsync -a "${BOOT}"/ "${MOUNT_POINT}"
+
+    umount "${MOUNT_POINT}"
+
+    mount "${LOOP_DEV}p2" "${MOUNT_POINT}"
 
-info "formatting boot partition"
-mkfs.vfat "${boot_partition}" 1>&2-
+    rsync -apS "${R}"/ "${MOUNT_POINT}"
 
-info "formatting rootfs partition"
-mkfs.ext4 "${root_partition}" 1>&2-
+    ;;
+"${IMAGE_FOR_RPI_WITH_MENDER_CLIENT}")
+    image_size="$(create_image 16 fat32:100 "${total}" "${total}" "${MENDER_DATA_SIZE}")"
+    info "${IMAGE} of size ${image_size}M was successfully created"
 
-# The root partition is expanded to fit the SD card. However, the size of the
-# SD card is currently unknown, so reserving 5% of the filesystem blocks at
-# this stage is incorrect.
-tune2fs -m 0 "${root_partition}"
+    info "creating loop device and scanning partition table"
 
-mount "${boot_partition}" "${MOUNT_POINT}"
+    scan_partition_table
 
-rsync -a "${BOOT}"/ "${MOUNT_POINT}"
+    info "${LOOP_DEV} is successfully created"
 
-umount "${MOUNT_POINT}"
+    info "formatting partitions"
 
-mount "${root_partition}" "${MOUNT_POINT}"
+    format_partitions vfat ext4 ext4 ext4
 
-rsync -apS "${R}"/ "${MOUNT_POINT}"
+    mount "${LOOP_DEV}p1" "${MOUNT_POINT}"
+
+    rsync -a "${BOOT}"/ "${MOUNT_POINT}"
+
+    cp "${TOOLSET_DIR}"/mender/boot.scr "${MOUNT_POINT}"
+    cp "${TOOLSET_DIR}"/mender/u-boot.bin "${MOUNT_POINT}"/kernel7.img
+    sed -i -e "s/\b[ ]root=[^ ]*/ root=\${mender_kernel_root}/" "${MOUNT_POINT}"/cmdline.txt
+
+    umount "${MOUNT_POINT}"
+
+    mount "${LOOP_DEV}p2" "${MOUNT_POINT}"
+
+    rsync -apS "${R}"/ "${MOUNT_POINT}"
+
+    install_mender
+
+    umount "${MOUNT_POINT}"
+
+    # The data partition has to contain /mender/device_type and
+    # /u-boot/fw_env.config
+    mount "${LOOP_DEV}p4" "${MOUNT_POINT}"
+    mkdir "${MOUNT_POINT}"/mender
+    mkdir "${MOUNT_POINT}"/u-boot
+    install -m 0644 "${PIEMAN_DIR}"/files/mender/device_type "${MOUNT_POINT}"/mender
+    install -m 0644 "${PIEMAN_DIR}"/files/mender/fw_env.config "${MOUNT_POINT}"/u-boot
+
+    ;;
+"${IMAGE_ARTIFACT}")
+    dd if=/dev/zero of="${IMAGE}" bs="$((1024 * 1024))" seek="${total}" count=1
+
+    LOOP_DEV="$(losetup -f)"
+
+    losetup "${LOOP_DEV}" "${IMAGE}"
+
+    mkfs.ext4 "${LOOP_DEV}"
+
+    mount "${LOOP_DEV}" "${MOUNT_POINT}"
+
+    rsync -apS "${R}"/ "${MOUNT_POINT}"
+
+    install_mender
+
+    umount "${MOUNT_POINT}"
+
+    info "converting ${IMAGE} into artifact"
+
+    "${TOOLSET_DIR}"/mender/mender-artifact write rootfs-image \
+        --update "${IMAGE}" \
+        --output-path "${ARTIFACT}" \
+        --artifact-name "${MENDER_ARTIFACT_NAME}" \
+        --device-type raspberrypi3
```

Comment:
What if we create a 'mapper' which will map our `DEVICE` parameter into this value? The same solution for the files/mender/device_type file